src/dense_op.cc

```diff
@@ -0,0 +1,190 @@
+#include "dense_op.hh"
+
+#include <iostream>
+#include <stdio.h>
+#include <omp.h>
+
+#include <math.h>
+#include <typeinfo>
+
+#include <Random123/philox.h>
+#include <Random123/boxmuller.hpp>
+#include <Random123/uniform.hpp>
+
+
+namespace RandBLAS::dense_op {
+
+/*
+Note from Random123: Simply incrementing the counter (or key) is effectively indistinguishable
+from a sequence of samples of a uniformly distributed random variable.
+
+Notes:
+1. Currently, Not sure how to make generalized implementations to support both 2 and 4 random number generations.
+2. Note on eventually needing to add 8 and 16-bit generations.
+*/
+
+
+
+// Actual work - uniform dirtibution
+template <typename T, typename T_gen>
+static void gen_unif(int64_t n_rows, int64_t n_cols, T* mat, uint32_t seed)
+{
+        typename T_gen::key_type key = {{seed}};
```

Comment:
A couple things: I think we should take in a pointer to a uint32_t as a seed, use it as a counter (rather than a key), and increment it as we go along. It would also be good for us to accept a key separately from the seed. That way we could support embarrassingly parallel computations. The semantics would always be that "seed" is passed as a pointer and incremented while "key" is not.